runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/ExecutableStageDoFnOperator.java

```diff
@@ -222,12 +189,37 @@ protected void addSideInputValue(StreamRecord<RawUnionValue> streamRecord) {
 
   private class SdkHarnessDoFnRunner implements DoFnRunner<InputT, OutputT> {
     @Override
-    public void startBundle() {}
+    public void startBundle() {
+      checkState(
+          stageBundleFactory != null, "%s not yet prepared", StageBundleFactory.class.getName());
+      checkState(
+          stateRequestHandler != null, "%s not yet prepared", StateRequestHandler.class.getName());
+      OutputReceiverFactory receiverFactory =
+          new OutputReceiverFactory() {
+            @Override
+            public FnDataReceiver<OutputT> create(String pCollectionId) {
+              return (receivedElement) -> {
+                // handover to queue, do not block the grpc thread
+                outputQueue.put(KV.of(pCollectionId, receivedElement));
+              };
+            }
+          };
+
+      try {
+        remoteBundle =
+            stageBundleFactory.getBundle(receiverFactory, stateRequestHandler, progressHandler);
+      } catch (Exception e) {
+        throw new RuntimeException(e);
+      }
+    }
 
     @Override
-    public void processElement(WindowedValue<InputT> elem) {
+    public void processElement(WindowedValue<InputT> element) {
       try {
-        processElementWithSdkHarness(elem);
+        checkState(remoteBundle != null, "%s not yet prepared", RemoteBundle.class.getName());
```

Comment:
Just for my education, why the convention to print the classname when a field is not set? (Also above.)

Reply:
Just what the original author decided to do, not a convention.